File: datasets/IR-Plag/convertai-original-java-rest-typescript/case-07/non-plagiarized/06/T07.ts
import * as readline from 'readline';

/**
 *
 * @author 92E0988C1682C76D4D307AA15EC8346E
 */
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

function sumMajorDiagonal(m: number[][]): number {
  let result = 0;
  for (let i = 0; i < 4; i++) {
    result += m[i][i];
  }
  return result;
}

rl.question('Enter a 4-by-4 matrix row to row: ', (input) => {
  const matrix: number[][] = [];
  const rows = input.trim().split('\n');
  for (const row of rows) {
    matrix.push(row.trim().split(' ').map(Number));
  }

  const result = sumMajorDiagonal(matrix);
  console.log(`Sum of the element in the major diagonal is ${result}`);

  rl.close();
});

File: datasets/IR-Plag/convertai-original-java-rest-python/case-07/non-plagiarized/06/T07.py
import numpy as np

def sum_major_diagonal(matrix):
    result = 0
    for i in range(4):
        result += matrix[i][i]
    return result

if __name__ == "__main__":
    matrix = np.zeros((4, 4))
    print("Enter a 4-by-4 matrix row to row: ")
    for i in range(4):
        for j in range(4):
            matrix[i][j] = float(input())
    result = sum_major_diagonal(matrix)
    print(f"Sum of the element in the major diagonal is {result}")

